### PR TITLE
Add python version of step-49

### PR DESCRIPTION
**Pull request by agrayver**
Well, I underestimated the time and effort required for this, but here it is: step-49 in python. 

I took the liberty to classify this as a major change since the cumulative amount of required coding to enable this was substantial.

Here is the direct [link](https://github.com/agrayver/dealii/blob/add_step_49_python/examples/step-49/step-49.ipynb) to the rendered notebook.

Blocked by #9510 

Part of #9015

### CODE REVIEW

**Comment by peterrum** [comment]
Very cool :)

File: examples/step-49/doc/intro.dox [hunk 1]
```diff
@@ -6,6 +6,10 @@ were contributed by Yuhan Zhou, Wolfgang Bangerth, and David Wells.</i>
 This tutorial is an extension to step-1 and demonstrates several ways to
 obtain more involved meshes than the ones shown there.
 
+@note This tutorial is also available as a Jupyter Python notebook that 
+  uses the deal.II python interface. The notebook is available in the 
+  same directory as the original C++ program.
```
Comment (Rombur) [inline examples/step-49/doc/intro.dox]: Can you move the Python tutorial with the others Python tutorials instead? I think that's what we decided last time.

File: examples/step-49/step-49.ipynb [hunk 2]
```diff
@@ -0,0 +1,1128 @@
...
+    "        if color_scheme:\n",
+    "            colors.append(color_scheme(cell))\n",
+    "\n",
+    "        cell_id += 1\n",
+    "    p = PatchCollection(patches, edgecolors='k', facecolors=None)  \n",
+    "    p.set_array(np.array(colors))\n",
+    "    ax.add_collection(p, autolim=True)\n",
+    "    ax.autoscale_view()\n",
+    "    ax.set_aspect(1)\n",
+    "    plt.show()"
```
Comment (Rombur) [inline examples/step-49/step-49.ipynb]: Can you add the image.

File: examples/step-49/step-49.ipynb [hunk 3]
```diff
@@ -0,0 +1,1128 @@
...
+   "metadata": {},
+   "source": [
+    "## Introduction"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This tutorial is an extension to [step-1](https://www.dealii.org/current/doxygen/deal.II/step_49.html) and demonstrates several ways to obtain more involved meshes than the ones shown there.\n",
```
Comment (Rombur) [inline examples/step-49/step-49.ipynb]: Don't you want to link to step-1 instead of step-49?

File: examples/step-49/step-49.ipynb [hunk 4]
```diff
@@ -0,0 +1,1128 @@
...
+   ],
+   "source": [
+    "help(dealii.Triangulation.read)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Meshes can be generated from different tools like [gmsh](http://gmsh.info/), [lagrit](https://lagrit.lanl.gov/) and [cubit](https://cubit.sandia.gov/). See the documentation of [GridIn](https://www.dealii.org/current/doxygen/deal.II/classGridIn.html) for more information. The problem is that deal.II needs meshes that only consist of quadrilaterals and hexahedra – tetrahedral meshes won't work (this means tools like tetgen can not be used directly).\n",
```
Comment (Rombur) [inline examples/step-49/step-49.ipynb]: I've never heard of lagrit before. Looks pretty nice. Did you know about this @jppelteret ?

File: examples/step-49/step-49.ipynb [hunk 5]
```diff
@@ -0,0 +1,1128 @@
...
...a1bt2aRCIRbdu2jV2vUqnI0tKyxRADX2RmZpK+vj516dKFVCoVubm50ejRo5v08c0331CPHj1IIpGQQCAgX19f2rBhA9XU1LB28+fPJ7FY3OJDiYioW7du1L1792br3dzcaMKECVqPz87OjmbPnq11u0KhIABNjuVVqFQqAtBEDqLx/pcvX6ZPP/2UhEIhtWrVigwNDdmwqLm5OXXs2JEmTZpE+/fvp6FDh3JKUBC1LEXyKv62Dv9NL/8Jh9+S8l5jXL58mZfD56tuOXjwYOJzXp06daKBAwdy2nXo0IHCwsI47QYOHEidO3fmtDt06BDp6+tz2hE1KFpqnEBRURFFR0dT3759qVWrVgSA9PT0yNPTk9577z1asGABL62hV9GzZ88WHYIGWVlZJJVKqU+fPpSXl0fm5uaUkZHx2v38WURFRbG/g0qlog0bNpClpSUJBALq3bs33b9/n2JiYkgoFFJJSQmvNhMSEohhGDp16pROu0WLFpFAIKCAgAAqKyuj2bNnk0QioYqKij90LllZWSSTyahDhw7sCHXChAnk5OSkdZ+ffvqJ+vXrRzKZjBiGIXd3d5o6dSoBoN27d2vdz8jIiNatW9dsfUBAgM6BjK+vLw0fPlznefAZ0AGg9evX0wcffEA9evQgBwcHdsQuFArJzMyMGIYhb29v2rJli9Z5l4kTJ5Kbm5vOvogaHH5SUpJOmzfp8P/JtOWIQfKNzf/V/HqlUsmrzGBdXZ1OrXENKisrecXb8/LyeJc3LCkpYemBZmZmmDVrFs6ePYuCggLU1dXhq6++go+PD+Lj47F+/XpUV1cjIiKCU+e+MTp27IjMzMwWtxUXF6N9+/Zwd3fHuXPnYGVlhcjISCxZsoR3+38FiouLsXbtWqxduxZAwzU1d+5c5OfnIy4uDrm5ufD29sb777+P/v3788pxUCqVGD58OEJDQzFw4ECdtp9//jnu3r2Lp0+fwsLCAps3b0Z0dDSnLEVLyMnJgbe3N9zc3HDjxg32mo6IiEBOTo7W/QYNGoSff/4ZVVVVuHTpEtq2bYtdu3YBAEsvvXTpUpN9CgsLUVFR0WLhE2tra506N5aWlpxF7kUiEZ48eYKUlBTs3LkT06ZNQ+/eveHq6gq5XM7eh4sXL0ZcXByABgrr7t27kZubC6VSydYlHj58OGbOnKl13kUqlfKa72MYhpc0xJ
```
Comment (Rombur) [inline examples/step-49/step-49.ipynb]: Typo `next*`

Comment (Rombur) [inline examples/step-49/step-49.ipynb]: This line is not necessary

Comment (Rombur) [inline examples/step-49/step-49.ipynb]: This line is not necessary

Comment (Rombur) [inline examples/step-49/step-49.ipynb]: Not necessary

Comment (Rombur) [inline examples/step-49/step-49.ipynb]: Not necessary

**Comment by Rombur** [comment]
You need to run the indentation script.

File: examples/step-49/doc/intro.dox (hunk 1 above)
Comment (agrayver) [reply examples/step-49/doc/intro.dox]: Shall follow the *step-62* and add a symbolic link to the original notebook in the contrib dir? This way I can use the files from the step-49's directory (e.g. example.msh).

File: examples/step-49/step-49.ipynb (hunk 5 above)
Comment (agrayver) [reply examples/step-49/step-49.ipynb]: You mean you don't want the rendered images to be present in the notebook?

File: examples/step-49/doc/intro.dox (hunk 1 above)
Comment (Rombur) [reply examples/step-49/doc/intro.dox]: Yes that's fine too

File: examples/step-49/step-49.ipynb (hunk 5 above)
Comment (Rombur) [reply examples/step-49/step-49.ipynb]: Hmm I don't know why all my comments are on the line 337, looks like a bug in github. Several times you do something like:
```python
triangulation = Triangulation('2D')
triangulation = create_triangulation()
```
The first line is useless.

**Comment by agrayver** [comment]
@Rombur I addressed your comments.

**Comment by agrayver** [comment]
@Rombur ok, done!

**Pull request merged**